File: plugins/compiled/grass_block.js
import { game, player } from "./lib.runtime.js"
import * as random from "./random.js"

export var thisBlock = 0

export class myBlock {
	title = "None"
	constructor(title) {
		this.title = title
	}

}
export const HAS_BEEN_PRESSED = false

export function place() {
	player.inventory.take("block/chest")
	game.block.setBlock("chest:unopened")
}

export function interact() {
	let amount = random.randint(1, 3)
	if(player.inventory.has("item/key") == false) return
	player.inventory.take("item/key")
	player.inventory.give("item/diamond", amount)
	game.block.setBlock("chest:unopened")

	return Math.abs(100)
}

File: plugins/grass_block.py
from lib.runtime import game, player
import random

thisBlock = 0

class myBlock:
	title = "None"
	def __init__(self, title):
		self.title = title

HAS_BEEN_PRESSED = False

def place():
	player.inventory.take("block/chest")
	game.block.setBlock("chest:unopened")

def interact():
	amount = random.randint(1, 3)
	if(player.inventory.has("item/key") == False): return
	player.inventory.take("item/key")
	player.inventory.give("item/diamond", amount)
	game.block.setBlock("chest:unopened")

	return abs(100)